syntax = "proto3";
package et;

enum PacketHeader {
  EMPTY = 0;
  INSERT_KEYS = 1;
  INIT_STATE = 2;
  CLIENT_CLOSE_PANE = 3;
  APPEND_TO_PANE = 4;
  NEW_TAB = 5;
  CONSOLE_KEYS = 6;
  APPEND_TO_CONSOLE = 7;
  SERVER_CLOSE_PANE = 8;
}

message Pane {
  string id = 1;
  bool active = 2;
}

message SplitPane {
  bool vertical = 1;
  repeated PaneOrSplitPane panes = 2;
  repeated float sizes = 3;
}

message PaneOrSplitPane {
  Pane pane = 1;
  SplitPane split_pane = 2;
}

message Tab {
  string id = 1;
  int32 order = 2;
  bool active = 3;
  PaneOrSplitPane pane = 4;
}

message State {
  repeated Tab tab = 1;
  string shell = 2;
}
